peer.go

```diff
@@ -1495,19 +1496,24 @@ func (p *peer) ChannelSnapshots() []*channeldb.ChannelSnapshot {
 	return snapshots
 }
 
-// genDeliveryScript returns a new script to be used to send our funds to in
+// genDeliveryScript returns a script to be used to send our funds to in
 // the case of a cooperative channel close negotiation.
-func (p *peer) genDeliveryScript() ([]byte, error) {
-	deliveryAddr, err := p.server.cc.wallet.NewAddress(
-		lnwallet.WitnessPubKey, false,
-	)
-	if err != nil {
-		return nil, err
+func (p *peer) genDeliveryScript(deliveryAddr *btcutil.Address) ([]byte, error) {
```

Comment:
Instead of modifying this method, is it possible to always pass in a valid `deliveryAddr` to `CloseLink`?

Reply:
I changed it such that the delivery address is always passed to CloseLink. It pushed the responsibility of determining the address to the callers, which is more "natural" for the rpcserver, but somehow less for the calls from the wrapper closeLink in server.go. Let me know if you think its indeed better this way or you want me to revert to the previous approach. Thanks!